source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,1896 @@
+      updates: [ { q: {}, u: { $set: { x: 1 } } } ]
+      ordered: true
+      writeConcern: { $$exists: false }
+
+
+$$type
+``````
+
+Syntax, where ``bsonType`` is a string or integer::
+
+    { $$type: <bsonType> }
+    { $$type: [ <bsonType>, <bsonType>, ... ] }
+
+This operator can be used anywhere a matched value is expected (including an
+`expectedResult <operation_expectedResult_>`_). The test runner MUST assert that
+the actual value exists and matches one of the expected types, which correspond
+to the documented types for the
+`$type <https://docs.mongodb.com/manual/reference/operator/query/type/>`__
+query operator.
+
+An example of this operator follows::
+
+    command:
+      getMore: { $$type: [ int, long ] }
+      collection: { $$type: 2 } # string
+
+When the actual value is an array, test runners MUST NOT examine types of the
+array's elements. Only the type of actual field should be checked. This is
+admittedly inconsistent with the behavior of the
+`$type <https://docs.mongodb.com/manual/reference/operator/query/type/>`__
+query operator, but there is presently no need for this behavior in tests.
+
+
+$$unsetOrMatches
+````````````````
+
+Syntax::
+
+    { $$unsetOrMatches: <anything> }
+
+This operator can be used anywhere a matched value is expected (including an
+`expectedResult <operation_expectedResult_>`_). The test runner MUST assert that
+actual value either does not exist or and matches the expected value. Matching
+the expected value should use the standard rules in `Evaluating Matches`_, which
+means that it may contain special operators.
+
+This operator is primarily used to assert driver-optional fields from the CRUD
+spec (e.g. ``insertedId`` for InsertOneResult, ``writeResult`` for
+BulkWriteException).
+
+An example of this operator used for a result's field follows::
+
+    expectedResult:
+      insertedId: { $$unsetOrMatches: 2 }
+
+An example of this operator used for an entire result follows::
+
+    expectedError:
+      expectedResult:
+        $$unsetOrMatches:
+          deletedCount: 0
+          insertedCount: 2
+          matchedCount: 0
+          modifiedCount: 0
+          upsertedCount: 0
+          upsertedIds: { }
+
+
+$$sessionLsid
+`````````````
+
+Syntax::
+
+    { $$sessionLsid: <string> }
+
+This operation is used for matching any value with the logical session ID of a
+`session entity <entity_session_>`_. The value will refer to a unique name of a
+session entity. The YAML file SHOULD use an `alias node`_ for a session entity's
+``id`` field (e.g. ``session: *session0``).
+
+An example of this operator follows::
+
+    command:
+      ping: 1
+      lsid: { $$sessionLsid: *session0 }
+
+
+Test Runner Implementation
+--------------------------
+
+The sections below describe instructions for instantiating the test runner,
+loading each test file, and executing each test within a test file. Test runners
+SHOULD NOT share state created by processing a test file with the processing of
```

Comment:
I think it will suffice to say 
> Test runners SHOULD NOT share state between tests.

Reply:
There is state that _can_ be shared between tests or test files.

The internal MongoClient is created per test file and shared by tests in that file (this might change if we move `allowMultipleMongoses`, after which it might be shared across _all_ test files). Likewise for the default database/collection names (which might also be removed). Another bit of global context would be server version/topology info, which the test runner might cache to optimize `runOnRequirement` checks.